test.sh

```diff
@@ -64,10 +95,108 @@ function run_test_coverage() {
+    -c, --coverage                        Adds coverage to the list of tests to run
+    -i, --integration                     Adds integration to the list of tests to run
+    -s, --start-py                        Adds start (py) to the list of tests to run
+    -v, --gomod-vendor                    Adds gomod-vendor to the list of tests to run
+    -g, --generate                        Adds generate to the list of tests to run
+    -r, --rpm                             Adds rpm to the list of tests to run
+    -p, --show-integration-test-list      Outputs a list of the available integration tests
+    -f, --integration-filter <REGEX>      Run only those tests and examples matching the regular expression
+
+                                          Note:
+                                           This option disables the '"back in time"' integration test setup
+
+                                           For tests, the regular expression is split by unbracketed slash (/)
+                                           characters into a sequence of regular expressions
+
+                                          Example:
+                                           TestAkamaiPurgerDrainQueueFails/TestWFECORS
+    -h, --help                            Shows this help message
+
+EOM
+)"
+
+while getopts luecipsvgrnhd:f:-: OPT; do
+  if [ "$OPT" = - ]; then   # long option: reformulate OPT and OPTARG
+    OPT="${OPTARG%%=*}"       # extract long option name
+    OPTARG="${OPTARG#$OPT}"   # extract long option argument (may be empty)
+    OPTARG="${OPTARG#=}"      # if long option argument, remove assigning `=`
+  fi
+  case "$OPT" in
+    l | lints )                      RUN+=("lints") ;;
+    u | unit )                       RUN+=("unit") ;;
+    d | unit-dir-filter )            check_arg; UNIT_FILTER+=("${OPTARG}") ;;
+    e | enable-race-detector )       TRAVIS="true" ;;
+    c | coverage )                   RUN+=("coverage") ;;
+    i | integration )                RUN+=("integration") ;;
+    p | show-integration-test-list ) print_list_of_integration_tests ;;
+    f | integration-filter )         check_arg; INT_FILTER+=("--filter" "${OPTARG}") ;;
+    s | start )                      RUN+=("start") ;;
+    v | gomod-vendor )               RUN+=("gomod-vendor") ;;
+    g | generate )                   RUN+=("generate") ;;
+    r | rpm )                        RUN+=("rpm") ;;
+    n | conf-next )                  export BOULDER_CONFIG_DIR="test/config-next" ;;
+    h | help )                       print_usage_exit ;;
+    ??* )                            exit_msg "Illegal option --$OPT" ;;  # bad long option
+    ? )                              exit 2 ;;  # bad short option (error reported via getopts)
+  esac
+done
+shift $((OPTIND-1)) # remove parsed options and args from $@ list
+
+# The list of segments to run. To run only some of these segments, pre-set the
+# RUN variable with the ones you want (see .travis.yml for an example).
```

Comment:
Although this script does currently preserve backwards compatibility with the env var approach, I wouldn't document such here. The run only some of the segments, use the flags.